File: scm-csharp/scheme/primitives/PrimitiveBaseName.cs
namespace scheme;

public class PrimitiveBaseName : Primitive
{
    public override string Name()
    {
        return "base-name";
    }

    public override string Info()
    {
        return
            "Syntax: (base-name path)\n" +
            "Library: (scm system)\n" +
            "Description: Returns the file name (including extension) from the given path string, without the directory part.\n" +
            "Example:\n" +
            "  (base-name \"/usr/share/doc/readme.txt\") => \"readme.txt\"";
    }
    
    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        var path = new String(Value.AsString(arguments[0]));
        return Path.GetFileName(path).ToCharArray();
    }
}

File: scm-java/src/main/scheme/primitives/PrimitiveBaseName.java
package scheme.primitives;

import scheme.*;

import java.io.File;

public class PrimitiveBaseName extends Primitive
{
    @Override
    public String name() {
        return "base-name";
    }

    @Override
    public String info() {
        return "Syntax: (base-name path)\n" +
               "Library: (scm fs)\n" +
               "Description: Returns the file name (including extension) from the given path string, without the directory part.\n" +
               "Example:\n" +
               "  (base-name \"/usr/share/doc/readme.txt\") => \"readme.txt\"";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        var path = new String(Value.asString(arguments[0]));
        return new File(path).getName().toCharArray();
    }
}
